src/stores/FlairStore.js

```diff
@@ -149,15 +152,22 @@ class FlairStore {
             return this._groupProfiles[groupId];
         }
 
-        const profile = await matrixClient.getGroupProfile(groupId);
+        // No request yet, start one
+        if (!this._groupProfilesPromise[groupId]) {
+            this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId);
+        }
+
+        const profile = await this._groupProfilesPromise[groupId];
```

Comment:
you might want to think about what happens if the `getGroupProfile` request fails. I think you'll get stuck with a failed promise, which is sad-making.

Reply:
Actually, the error will be thrown as normal

Reply:
yes, but then every attempt to get the profile for this group will then fail immediately for the next half-hour, even if whatever network blip made the first request fail is fixed.

Reply:
Ah okay, I shall wrap it with a try/catch and delete the failed promise in the catch.